core/blockchain/impl/cached_tree.hpp

```diff
@@ -80,8 +82,15 @@ namespace kagome::blockchain {
 
     bool operator==(const TreeNode &other) const;
     bool operator!=(const TreeNode &other) const;
+    bool operator<(const TreeNode &other) const;
   };
 
+  inline bool operator<(const std::shared_ptr<TreeNode> &lhs,
```

Comment:
No, it does not. This binary op compare sptrs by their object

Reply:
It's ambigous because `std::shared_ptr` already has comparator.
Can they conflict because `std` comparator is visible by ADL, but tihs overload not visible?

This is `DerefLess` comparator.